lib/request/complement.js

```diff
@@ -0,0 +1,30 @@
+var hasIntersection = require('falcor-path-utils').hasIntersection;
+
+/**
+ * creates the complement of the requested and optimized paths
+ * based on the provided tree.
+ *
+ * If there is no complement then the this is just a glorified
```

Comment:
I bet i could use some sort of flag and `slice` to improve the performance where if there is no complement then its a noop `n` loop.